tap_google_sheets/singer_transform.py

```diff
@@ -0,0 +1,406 @@
+
+        if self.integer_datetime_fmt == NO_INTEGER_DATETIME_PARSING:
+            return string_to_datetime(value)
+        else:
+            try:
+                if self.integer_datetime_fmt == UNIX_SECONDS_INTEGER_DATETIME_PARSING:
+                    return unix_seconds_to_datetime(value)
+                else:
+                    return unix_milliseconds_to_datetime(value)
+            except:
+                return string_to_datetime(value)
+
+    def _transform(self, data, typ, schema, path):
+        if self.pre_hook:
+            data = self.pre_hook(data, typ, schema)
+
+        if typ == "null":
+            if data is None or data == "":
+                return True, None
+            else:
+                return False, None
+
+        elif schema.get("format") == "date-time":
+            data = self._transform_datetime(data)
+            if data is None:
+                return False, None
+
+            return True, data
+        elif schema.get("format") == "singer.decimal":
+            if data is None:
+                return False, None
+
+            if isinstance(data, (str, float, int)):
+                try:
+                    return True, str(decimal.Decimal(str(data)))
+                except:
+                    return False, None
+            elif isinstance(data, decimal.Decimal):
+                try:
+                    if data.is_snan():
+                        return True, 'NaN'
+                    else:
+                        return True, str(data)
+                except:
+                    return False, None
+
+            return False, None
+        elif typ == "object":
+            # Objects do not necessarily specify properties
+            return self._transform_object(data,
+                                          schema.get("properties", {}),
+                                          path,
+                                          schema.get(SchemaKey.pattern_properties))
+
+        elif typ == "array":
+            return self._transform_array(data, schema["items"], path)
+
+        elif typ == "string":
+            if data is not None:
+                try:
+                    return True, str(data)
+                except:
+                    return False, None
+            else:
+                return False, None
+
+        elif typ == "integer":
+            if isinstance(data, str):
+                data = data.replace(",", "")
+
+            try:
+                return True, int(data)
+            except:
+                return False, None
+
+        elif typ == "number":
+            if isinstance(data, str):
+                data = data.replace(",", "")
+            try:
+                return True, float(data)
+            except:
+                return False, None
+
+        elif typ == "boolean":
+            if isinstance(data, str) and data is not None:
+                try:
+                    return True, str(data)
+                except:
```

Comment:
The `isinstance(data, str)` is already checking the data type of `data` so is there any chance to get an exception in `str(data)`? If not then return `True, data` directly without try-except because `data` is already in str form.